src/PanZoom.js

```diff
@@ -131,13 +131,19 @@ class PanZoom extends React.Component<Props> {
   }
 
   onDoubleClick = (e) => {
-    const { doubleZoomSpeed } = this.props
+    const { doubleZoomSpeed, onDoubleClick } = this.props
+    if (onDoubleClick !== undefined) {
```

Comment:
I think we should use `typeof onDoubleClick === 'function'` here

Reply:
I can do that, but keep in mind that the functionality is different.

```jsx
// doule click will result in nothing
const example1 = <PanZoom onDoubleClick={null} />
// doule click will result in error
const example2 = <div onDoubleClick={null} />
```

Might be just nitpicking but ReactDOM does not do a `typeof` check when calling an event handler. It just checks whether the property is `undefined` or not.

We should at least add a warning in case one of the handlers is not a function if we do that?

What is your opinion on this, @mnogueron ?

Reply:
OK I just checked again, this is only the case with react-native, not ReactDOM.

In case the passed prop is not a function you will get the following warning:

```
Warning: Expected `onDoubleClick` listener to be a function, instead got a value of `object` type.
```

I will adjust that accordingly.

Reply:
```js-jsx
// double click will result in nothing
const example1 = <PanZoom onDoubleClick={null} />
// double click will result in error
const example2 = <div onDoubleClick={null} />
```
Actually your first example will result in an error as checking for `undefined` is not checking for `null`. You will run `onDoubleClick(e)` which will be `null(e)` and that results in an error.
Checking for a function is not so much about forcing the user to pass a function, but for us to be able to safely do `onDoubleClick(e)`.

Reply:
It should now behave like plain react (https://codesandbox.io/s/funny-darwin-1zn2e?fontsize=14)